Objects/moduleobject.c

```diff
@@ -739,6 +739,29 @@ _PyModuleSpec_IsInitializing(PyObject *spec)
     return 0;
 }
 
+/* Check if the submodule name is in the "_uninitialized_submodules" attribute
+   of the module spec.
+   Clear the exception and return 0 if spec is NULL.
+ */
+int
+_PyModuleSpec_IsUninitializedSubmodule(PyObject *spec, PyObject *name)
+{
+    if (spec != NULL) {
```

Comment:
I prefer to avoid nesting the conditionals so every case is more clear:

```suggestion
{
    if (spec == NULL) {
         goto exit;
    }
      
    _Py_IDENTIFIER(_uninitialized_submodules);
    PyObject *value = _PyObject_GetAttrId(spec, &PyId__uninitialized_submodules);
     if (value == NULL) {
         goto error;
     }
     int is_uninitialized = PySequence_Contains(value, name);
     Py_DECREF(value);
     if (is_uninitialized == -1) {
         goto error
     }
     return is_uninitialized;
exit:
    PyErr_Clear();
    return 0;
}
```